core/mempool/src/lib.rs

```diff
@@ -1,143 +1,142 @@
-use futures::{Future, Stream};
-use log::error;
 use primitives::chain::{ChainPayload, ReceiptBlock, SignedShardBlock};
-use primitives::hash::hash_struct;
 use primitives::merkle::verify_path;
 use primitives::transaction::{verify_transaction_signature, SignedTransaction};
-use shard::ShardBlockChain;
+use primitives::hash::hash_struct;
+use storage::{ShardChainStorage, Trie, TrieUpdate, GenericStorage};
+use node_runtime::state_viewer::TrieViewer;
 use std::collections::HashSet;
-use std::sync::Arc;
-use std::sync::RwLock;
-use tokio::sync::mpsc::{Receiver, Sender};
+use std::sync::{Arc, RwLock};
 
-const POISONED_LOCK_ERR: &str = "The lock was poisoned";
-const TOKIO_RECV_ERR: &str = "Implementation of tokio Receiver should not return an error";
+const POISONED_LOCK_ERR: &str = "The lock was poisoned.";
 
 /// mempool that stores transactions and receipts for a chain
-struct Pool {
-    transactions: HashSet<SignedTransaction>,
-    receipts: HashSet<ReceiptBlock>,
-    // TODO: make it work for beacon chain as well
-    chain: Arc<ShardBlockChain>,
+pub struct Pool {
+    transactions: RwLock<HashSet<SignedTransaction>>,
```

Comment:
hm, we currently have lock on the whole client, what these locks adding to that?

Reply:
Oh I didn't realize that :(

Reply:
Actually in network we don't have lock on client